src/platform_impl/macos/event_loop.rs

```diff
@@ -13,25 +13,35 @@ use crate::{
     event::Event,
     event_loop::{ControlFlow, EventLoopClosed, EventLoopWindowTarget as RootWindowTarget},
     monitor::MonitorHandle as RootMonitorHandle,
-    platform_impl::platform::{
-        app::APP_CLASS,
-        app_delegate::APP_DELEGATE_CLASS,
-        app_state::AppState,
-        monitor::{self, MonitorHandle},
-        observer::*,
-        util::IdRef,
+    platform_impl::{
+        self,
+        platform::{
+            app::APP_CLASS,
+            app_delegate::{APP_DELEGATE_CLASS, APP_DELEGATE_CLASS_WITH_FILE_OPEN},
+            app_state::AppState,
+            monitor::{self, MonitorHandle},
+            observer::*,
+            util::IdRef,
+        },
     },
 };
 
 pub struct EventLoopWindowTarget<T: 'static> {
     pub sender: mpsc::Sender<T>, // this is only here to be cloned elsewhere
     pub receiver: mpsc::Receiver<T>,
+    pub delegate: IdRef,
+    pub delegate_with_file_open: IdRef,
 }
```

Comment:
I wasn't entirely sure about adding the application delegates here, but this was the only reasonable way I could make these accessible at `set_file_open_callback()`.